pinc/Project.inc

```diff
@@ -258,6 +259,25 @@ class Project
         return $errors;
     }
 
+    public function delete()
+    {
+        exec("rm -rf $this->dir");
```

Comment:
do we need to worry about shell escape here? I guess using php to delete the directory is probably slow?
https://stackoverflow.com/questions/3349753/delete-directory-with-files-in-it

Reply:
UGH. Given how powerful this command is, yes, yes we should. Thanks for pointing it out -- I'll add some better checks and escaping here.

Reply:
Added several more checks:
1. Validate `$this->projectID` to ensure that we aren't pointing to the top-level project directory (which would result in deleting *ALL* projects 😮).
2. Check the directory exists before trying to delete it.
3. Properly escape the directory argument as one final layer of protection (good idea Chris!).